pandas/core/tools/datetimes.py

```diff
@@ -316,8 +317,23 @@ def _convert_listlike_datetimes(
     elif unit is not None:
         if format is not None:
             raise ValueError("cannot specify both format and unit")
-        arg = getattr(arg, "values", arg)
-        result, tz_parsed = tslib.array_with_unit_to_datetime(arg, unit, errors=errors)
+        arg = getattr(arg, "_values", arg)
+        # GH 30050 pass an ndarray to tslib.array_with_unit_to_datetime
+        # because it expects an ndarray argument
+        if isinstance(arg, IntegerArray):
```

Comment:
ok this soln looks ok. wonder if we pass the values AND the mask directly to ```tslib.array_with_unit_to_datetime``` (pass an np.zeros mask for the non-integer NA case). would make this simpler from a code perspective (should be the same from perf).

this way all of the nan handling is in the same place.

Reply:
@jreback We'd need to add `mask` as an argument to `tslib.array_with_unit_to_datetime` (with ndarray type), then we'd fill it with zeros whenever a non-nullable int type is passed and introduce an `if` to check. 

However, in `datetimes.py` we'd need to get `_ndarray_values` from `IntegerArray` to pass the full ndarray of values to `tslib.array_with_unit_to_datetime` (which would rely on the implementation of non-public stuff in `IntegerArray`) or we'd need to pass the incomplete values array in `_convert_listlike_datetimes` and rebuild the full array in `tslib.array_with_unit_to_datetime`, which would differ from the approach taken with other nullable datatypes.

Do you see any other way to accomplish moving the nan-handling to `tslib.array_with_unit_to_datetime` or is one of these two approaches acceptable?

Reply:
i guess my comment was not clear

pass the values and mask to array_with_unit_to_datetime